### PR TITLE
Add path and target file dependency examples

### PR DESCRIPTION
**Pull request by jdub**
Fixes rust-lang/cargo#1684

### CODE REVIEW

File: src/doc/manifest.md [hunk 1]
```diff
@@ -128,6 +128,9 @@ git = "https://github.com/wycats/hammer.rs"
 
 [dependencies.color]
 git = "https://github.com/bjz/color-rs"
+
+[dependencies.geometry]
+path = "crates/geometry"
```
Comment (alexcrichton) [inline src/doc/manifest.md]: While you're at it, could you change these dependencies to the more concise inline table syntax?

File: src/doc/manifest.md [hunk 2]
```diff
@@ -157,6 +160,25 @@ openssl = "1.0.1"
 
 [target.x86_64-pc-windows-gnu.dependencies]
 winhttp = "0.4.0"
+
+[target.i686-unknown-linux-gnu.dependencies.native]
+path = "native/i686"
```
Comment (alexcrichton) [inline src/doc/manifest.md]: Could this (and the one below) use inline-table syntax instead?

**Comment by jdub** [comment]
What do you think about moving the crates.io section to the top and rephrasing it a little? It ought to be the common case, after all. Let me know; can do this in a follow-up.

**Comment by alexcrichton** [comment]
I think that the docs are slated for a much large overhaul soon, most likely, as they were written quite some time ago and could probably use a fresh "start from scratch" strategy. I believe @steveklabnik has been thinking about doing this at some point soon-ish!

Looks good to me, but could you revert the change to making the file executable as well? After that and a squash this should be good to go!

**Comment by jdub** [comment]
I managed to lose a commit during the squash, but I think I've sorted it now.

**Comment by alexcrichton** [comment]
@bors: r+ 5bbabe055593160395312c05007de09d4c5bce22

Thanks!

**Pull request merged**